### ISSUE TITLE
Beta6 - Panic in Prom

### ISSUE THREAD

**Issue by jranson** [Comcast/trickster#231]
```2019/05/30 07:56:50 http: panic serving 127.0.0.1:50996: runtime error: index out of range
goroutine 1690 [running]:
net/http.(*conn).serve.func1(0xc0001fbe00)
        /usr/local/Cellar/go/1.12.1/libexec/src/net/http/server.go:1769 +0x139
panic(0x15e8ec0, 0x1c2da40)
        /usr/local/Cellar/go/1.12.1/libexec/src/runtime/panic.go:522 +0x1b5
github.com/Comcast/trickster/internal/proxy/origins/prometheus.(*MatrixEnvelope).Crop(0xc00019a000, 0x0, 0xed481d115, 0x1c3e6e0, 0x0, 0xed481d81d, 0x1c3e6e0, 0xc0004cc9e0, 0xc0001f8b40)
        /Users/jranso000/code/go/src/github.com/Comcast/trickster/internal/proxy/origins/prometheus/matrix.go:119 +0x79e
github.com/Comcast/trickster/internal/proxy/engines.DeltaProxyCacheRequest(0xc000392c00, 0x17768c0, 0xc000020410, 0x17819c0, 0xc0000200f0, 0x177e3a0, 0xc000178410, 0x13a52453c000)
        /Users/jranso000/code/go/src/github.com/Comcast/trickster/internal/proxy/engines/deltaproxycache.go:231 +0xc3e
github.com/Comcast/trickster/internal/proxy/origins/prometheus.(*Client).QueryRangeHandler(0xc0000200f0, 0x17768c0, 0xc000020410, 0xc000218500)
        /Users/jranso000/code/go/src/github.com/Comcast/trickster/internal/proxy/origins/prometheus/handler_query_range.go:26 +0x2d7
net/http.HandlerFunc.ServeHTTP(0xc0000685b0, 0x17768c0, 0xc000020410, 0xc000218500)``

**Comment by jranson** [comment]
Pinned this one down to a collision because of using two versions of `i` in the nested loop (https://github.com/Comcast/trickster/blob/v1.0.6-beta6/internal/proxy/prometheus/matrix.go#L122). I have fixed in the 1.0-promsim branch.

**Comment by jranson** [comment]
Fixed in #232
